candump.c

```diff
@@ -688,8 +707,16 @@ int main(int argc, char **argv)
 				for (cmsg = CMSG_FIRSTHDR(&msg);
 				     cmsg && (cmsg->cmsg_level == SOL_SOCKET);
 				     cmsg = CMSG_NXTHDR(&msg,cmsg)) {
-					if (cmsg->cmsg_type == SO_TIMESTAMP)
+					if (cmsg->cmsg_type == SO_TIMESTAMP) {
 						memcpy(&tv, CMSG_DATA(cmsg), sizeof(tv));
+					} else if (cmsg->cmsg_type == SO_TIMESTAMPING) {
+
+						struct timespec *stamp = (struct timespec *)CMSG_DATA(cmsg);
+
+						stamp += 2;
```

Comment:
This offset calculation might be clear when looking at the referenced code or in the Linux code.
But "stamp += 2;" should be commented and properly referenced WHY we can find the hardware timestamp at this point.

Btw. thanks for the patch :-)